server/controllers/admin/nationalOccupancyController.test.ts

```diff
@@ -37,14 +46,23 @@ describe('NationalOccupancyController', () => {
     cruManagementAreaFactory.build({ id: CRU_AREA_WOMENS }),
   ]
   const nationalCapacity = cas1NationalOccupancyFactory.build()
+  const premisesCapacity = cas1PremiseCapacityFactory.build()
+
+  const premises = cas1PremisesFactory.build()
+
   const errorsAndUserInput = createMock<ErrorsAndUserInput>({ errors: {}, errorSummary: [] })
 
   let nationalOccupancyController: NationalOccupancyController
 
-  const mockRequest = (query: Record<string, string | Array<string>> = {}, session: Record<string, string> = {}) =>
+  const mockRequest = (
+    query: Record<string, string | Array<string> | number> = {},
+    session: Record<string, string | Array<string> | number> = {},
+    params: Record<string, string> = {},
```

Comment:
Could these types follow their actual implementation?

This seems to reveal a couple of inconsistencies with the session types, so would be worth tidying up, but would also remove the need for a couple of the castings further down.

```suggestion
    query: ParsedQs = {},
    session: NationalSpaceSearchFormData = {},
    params: ParamsDictionary = {},
```